### PR TITLE
fix: updating docker build pattern

### PR DESCRIPTION
**Pull request by drduhe**
**Issue #, if available:** n/a

### Notes
This PR introduces a significant refactor of the existing testing framework by modularizing the integration and load testing functionalities into separate, dedicated containers and codebases. This separation improves the clarity, maintainability, and scalability of our test suite. Additionally, new CLI scripts have been introduced to facilitate easier test execution .This refactor enhances the modularity and scalability of our testing framework, making it easier to manage and extend in the future. The introduction of dedicated containers and CLI scripts streamlines test execution, ensuring that each type of test can be run in its optimal environment.

**Key Changes:**

1. **Separation of Integration and Load Test Logic:**
   - **Integration Tests (`integ`)**:
     - Introduced a dedicated container specifically for running integration tests. This container focuses on end-to-end testing of the Tile Server, ensuring that all components work together as expected.
     - The integration test logic has been encapsulated within a new module `integ_processor.py`, which is responsible for setting up and executing the tests using `TestTileServer`.
     - A new class, `TSIntegTestProcessor`, extends `ProcessorBase` and manages the execution of integration tests, including configuration and error handling.
     - The `TSIntegTestProcessor` class leverages a `TSTestRequest` dataclass to capture and manage the parameters required for running integration tests.

   - **Load Tests (`load`)**:
     - A separate container has been created for load testing. This container is optimized for performance and stress testing, using Locust to simulate high user loads on the Tile Server.
     - Load testing logic is now contained within `load_processor.py`, a dedicated module that configures and executes load tests.
     - The `TSLoadTestProcessor` class, extending `ProcessorBase`, is introduced to manage the load test process, from setting up the environment to executing the tests.
     - A `TSLoadTestRequest` dataclass has been introduced to handle parameters for load tests, ensuring that all necessary configurations are passed correctly.

2. **Implementation of `ProcessorBase`:**
   - The common functionality for processing requests, handling success and failure messages, and interacting with AWS Lambda has been centralized in the `ProcessorBase` class.
   - Both `TSIntegTestProcessor` and `TSLoadTestProcessor` inherit from `ProcessorBase`, ensuring consistent handling of errors and responses across different types of tests.

3. **New CLI Scripts:**
   - **Integration Test CLI (`integ_cli.py`)**:
     - A new CLI script has been added to the `bin` directory for running integration tests. This script simplifies the execution of integration tests by accepting the container image URI as an argument and passing it to the `TSIntegTestProcessor`.
     - Usage example:
       ```bash
       python integ_cli.py --image_uri <image_uri>
       ```

   - **Load Test CLI (`load_cli.py`)**:
     - Another CLI script has been added to the `bin` directory for running load tests. This script supports a range of options for configuring the Locust load tests, including user count, run time, and spawn rate.
     - Usage example:
       ```bash
       python load_cli.py --image_uri <image_uri> --locust_users 10 --locust_run_time 10m
       ```

4. **Containerization:**
   - Both integration and load tests now have their own Docker containers, tailored to the specific requirements of each test type.
   - The integration test container includes all dependencies required for functional validation, while the load test container is equipped with Locust and its related tools for performance testing.

### Checklist

Before you submit a pull request, please make sure you have the following:
- [x] Code changes are compact and well-structured to facilitate easy review
- [x] Changes are documented in the README.md and other relevant documentation pages
- [x] PR title and description accurately reflect the changes and are detailed enough for historical tracking
- [x] PR contains tests that cover all new code and the code has been manual tested
- [x] All new dependencies are declared (if any), and no unnecessary libraries are added
- [x] Performance impacts (if any) of the changes are evaluated and documented
- [x] Security implications of the changes (if any) are reviewed and addressed
- [x] I have read the [Contributing Guidelines](../CONTRIBUTING.md) and agree to follow the [Code of Conduct](../CODE_OF_CONDUCT.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/aws/osml/tile_sever_test/processor_base.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        :param e: The exception that triggered the failure.
+        :returns: A dictionary with 'statusCode' set to 400 and a 'body' containing the error message and stack trace.
+        """
+        # Log the error and stack trace
+        stack_trace = traceback.format_exc()
+        logger.error(f"Error creating STAC items: {e}\nStack trace: {stack_trace}")
+
+        # Return the error message and stack trace in the response
+        error_response = {"message": str(e), "stack_trace": stack_trace.splitlines()}
+        return {"statusCode": 400, "body": json.dumps(error_response)}
```
Comment (markveerman) [inline src/aws/osml/tile_sever_test/processor_base.py:41]: Failure/Fault class exceptions are usually 500s whereas Error class client exceptions are 400s. You sure you want 400 here?

Comment (drduhe) [reply src/aws/osml/tile_sever_test/processor_base.py:41]: This is a good catch - this should probably be a 500 error.

**Comment by RanbirAulakh** [comment]
Since this is not a `chore`, it should be: `fix: update docker build pattern`

File: src/aws/osml/tile_server_test/utils/logger.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        root_logger.setLevel(level)
+    else:
+        logging.basicConfig(level=level)
+
+    return lambda_logger
+
+
+formatter = JsonFormatter(fmt="%(asctime)s %(name)s %(levelname)s %(image_hash)s %(message)s", datefmt="%Y-%m-%dT%H:%M:%S")
+
+filter = AsyncContextFilter(attribute_names=["image_hash"])
```
Comment (RanbirAulakh) [inline src/aws/osml/tile_server_test/utils/logger.py:103]: Update `image_hash` to `viewpoint_id`

File: src/aws/osml/tile_server_test/processor_base.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        logger.info(message)
+        return {"statusCode": 200, "body": json.dumps(message)}
+
+    @staticmethod
+    def failure_message(e: Exception) -> Dict[str, Any]:
+        """
+        Returns an error message in the form of a dictionary, including a stack trace, intended for an HTTP response.
+
+        :param e: The exception that triggered the failure.
+        :returns: A dictionary with 'statusCode' set to 400 and a 'body' containing the error message and stack trace.
```
Comment (RanbirAulakh) [inline src/aws/osml/tile_server_test/processor_base.py:33]: typo: 400 -> 500

File: src/aws/osml/tile_server_test/processor_base.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        """
+        # Log the error and stack trace
+        stack_trace = traceback.format_exc()
+        logger.error(f"Error creating STAC items: {e}\nStack trace: {stack_trace}")
```
Comment (RanbirAulakh) [inline src/aws/osml/tile_server_test/processor_base.py:37]: Wrong error message. This package does not correlate anything with STAC items. Either update `Error executing tile sever test .... etc`

File: src/aws/osml/tile_server_test/integ_processor.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        Initialize the processor with the runtime arguments.
+
+        :param event: The event dictionary containing runtime parameters.
+        """
+        self.request = TSTestRequest(**event)
+        self.s3_url = S3Url(self.request.image_uri)
+        self.test_config = TileServerIntegTestConfig(s3_bucket=self.s3_url.bucket, s3_key=self.s3_url.key)
+        self.ts_server = TestTileServer(self.test_config)
+
+    async def process(self) -> Dict[str, Any]:
```
Comment (RanbirAulakh) [inline src/aws/osml/tile_server_test/integ_processor.py:36]: Missing return type

File: setup.cfg [hunk 6]
```diff
@@ -39,12 +39,6 @@ python_requires = >=3.9
 include_package_data = True
 
 install_requires =
```
Comment (RanbirAulakh) [inline setup.cfg:41]: If this is empty, is this field needed?

Comment (drduhe) [reply setup.cfg:41]: Boilderplate code - left in other packages too.

**Comment by drduhe** [comment]
> Since this is not a `chore`, it should be: `fix: update docker build pattern`

Fixed

**Review by RanbirAulakh — APPROVED** [review]

**Pull request merged**